### PR TITLE
feat: allow sending notifications

### PR DESCRIPTION
**Pull request by cjdcordeiro**
- [x] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] I am following the [CONTRIBUTING](../CONTRIBUTING.md) guidelines

*Ping the @canonical/rocks team.*

---

## Description

Takes care of issuing notification whenever an event of interest occurs:
 - image processing failures: the corresponding workflow and job statuses are sent to the corresponding maintainer MM channels
 - documentation failures: same as above

A new workflow is also added to announce, via MM, on:
 - new releases
 - new packages (images) uploaded to GHCR

### CODE REVIEW

**Comment by cjdcordeiro** [comment]
Successful runs at https://github.com/cjdcordeiro/oci-factory/actions/runs/5279002416 and https://github.com/canonical/oci-factory/actions/runs/5276676973

Examples are in MM.

File: src/notifications/send_to_mattermost.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+if [ -z "$MM_CHANNEL_ID" ] || [ -z "$MM_BOT_TOKEN" ] || [ -z "$MM_SERVER" ]
+then
+    echo "ERROR: \$MM_CHANNEL_ID, \$MM_BOT_TOKEN and \$MM_SERVER must be provided"
+    exit 1
+fi
+
+# The message payload
+PAYLOAD_FILE="${PWD}/payload.json"
+
+# # What's the status of the job/pipeline that triggered this script?
```
Comment (SamirPS) [inline src/notifications/send_to_mattermost.sh:22]: typo # #

File: src/notifications/send_to_mattermost.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                "title": "${TITLE}", 
+                "title_link": "${URL}",
+                "footer": "${FOOTER}"
+            }
+        ]
+    }
+}
+EOF
+
+echo "Posting message to Mattermost channel ${MM_CHANNEL_ID}"
```
Comment (SamirPS) [inline src/notifications/send_to_mattermost.sh:53]: to the Mattermost channel no?

File: src/notifications/requirements.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash -ex
+
+YQ_VERSION=${YQ_VERSION:-"v4.32.2"}
+YQ_BINARY=${YQ_BINARY:-"yq_linux_amd64"}
+IMG_PATH=${IMG_PATH:-$PWD}
```
Comment (SamirPS) [inline src/notifications/requirements.sh:5]: why we need this ?

**Review by SamirPS — CHANGES_REQUESTED** [review]

File: src/notifications/send_to_mattermost.sh (hunk 1 above)
Comment (cjdcordeiro) [reply src/notifications/send_to_mattermost.sh:22]: ty, fixing

File: src/notifications/requirements.sh (hunk 3 above)
Comment (cjdcordeiro) [reply src/notifications/requirements.sh:5]: no need, just a leftover, good catch

File: src/notifications/send_to_mattermost.sh (hunk 2 above)
Comment (cjdcordeiro) [reply src/notifications/send_to_mattermost.sh:53]: both are acceptable, but what's actually missing is `'s`

File: src/notifications/send_to_mattermost.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                "fallback": "${SUMMARY}",
+                "text": "${SUMMARY}",
+                "color": "${COLOR}",
+                "title": "${TITLE}", 
+                "title_link": "${URL}",
+                "footer": "${FOOTER}"
+            }
+        ]
+    }
+}ramm
```
Comment (SamirPS) [inline src/notifications/send_to_mattermost.sh:50]: What is ramm ?

Comment (cjdcordeiro) [reply src/notifications/send_to_mattermost.sh:50]: wth

**Review by SamirPS — APPROVED** [review]

**Pull request merged**